lib/utils.pm

```diff
@@ -832,16 +832,29 @@ the second run will update the system.
 sub ssh_fully_patch_system {
     my $remote = shift;
     my $cmd_time = time();
+    my $resolver_option = get_var('PUBLIC_CLOUD_GEN_RESOLVER') ? ' --debug-solver ' : '';
+    my $cmd = "ssh $remote 'sudo zypper -n patch $resolver_option --with-interactive -l'";
     # first run, possible update of packager -- exit code 103
-    my $ret = script_run("ssh $remote 'sudo zypper -n patch --with-interactive -l'", 1500);
+    my $ret = script_run($cmd, 1500);
     record_info('zypper patch', 'The command zypper patch took ' . (time() - $cmd_time) . ' seconds.');
-    die "Zypper failed with $ret" if ($ret != 0 && $ret != 102 && $ret != 103);
-
+    if ($ret != 0 && $ret != 102 && $ret != 103) {
+        if (get_var('PUBLIC_CLOUD_GEN_RESOLVER')) {
```

Comment:
Kind proposal, being then `$resolver_option` already a checked value, in the next `if`s, isn't it better to check the value `if ($resolver_option)...` instead of repeating a `get(...)` function call ?

Reply:
I thought about this as well, but get_var is inexpensive after all.

Reply:
done